### PR TITLE
Fix #11505: Fileupload refactor messageTemplate

### LINKED ISSUE #11505: FileUpload: attribute messageTemplate has no effect anymore
### Describe the bug

Using the attribute messageTemplate in the fileUpload component does not change the way the added file message is formatted.

For instance, using 

```
    <p:fileUpload mode="advanced"
                  multiple="false"
                  messageTemplate="__{size} {name}__"
                  update="messages"
                  listener="#{testView.handleFileUpload}" />

```

i would expect that the resulting message prints the size and the name of the uploaded file enclosed by two underscores. Currently, is has no effect:

image

### Reproducer

Here is the repoducer:  [p-fileUpload.zip](https://github.com/primefaces/primefaces/files/14376252/p-fileUpload.zip)

### Expected behavior

_No response_

### PrimeFaces edition

None

### PrimeFaces version

14.0.0-RC1

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

3

### Java version

11

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #11505: Deprecate FileUpload messageTemplate

### CODE REVIEW

**Comment by melloware** [comment]
@tandraschko just double checking but i think the only reason the tests are now failing is it was counting on the widget holding these messsage but they were never being used and all of the display happens in the messages component now.  I ran the showcase with this code and all validations and errors appear to be happening.

File: primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js [hunk 1]
```diff
@@ -195,26 +186,8 @@ PrimeFaces.widget.FileUpload = PrimeFaces.widget.BaseWidget.extend({
             add: function(e, data) {
                 $this.chooseButton.removeClass('ui-state-hover ui-state-focus');
 
-                if($this.fileAddIndex === 0) {
-                    $this.clearMessages();
-                }
-
-                // we need to fake the filelimit as the jquery-fileupload input always only contains 1 file
-                var fileLimit = data.fileInput.data('p-filelimit');
```
Comment (tandraschko) [inline primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js]: Did you test filelimit with advanced fileupload?

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js]: Yep and it it stopped it at 3 when I tried to upload 4 files and the message was correct.

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js]: image

**Comment by melloware** [comment]
OK let me test some more today with simple mode.

**Comment by melloware** [comment]
OK it is use in Simple mode so I have done the following:

- Update the docs to reflect what `messageTemplate` actually does
- Only send `messageTemplate` in Simple mode now
- Updated MD docs to mark props deprecated with ~~strikethrough~~
- Update Taglib XML docs for those props

Next I think I need to fix the tests as the widget is no longer storing local messages.

**Comment by melloware** [comment]
OK @tandraschko @tuerker I got to the bottom of it.  `messageTemplate` is used in Simple mode to display the name and in Advanced mode only in a few validation scenarios which is why the Integration tests were failing.

So I have updated the docs accordingly.

**Comment by melloware** [comment]
Basically this whole PR is just a documentation update.

**Comment by melloware** [comment]
Agreed @tandraschko let me know when that other PR is ready so I can fix my PR after that one is merged?

**Pull request merged**